webapp-client/src/component/Navbar.js

```diff
@@ -2,6 +2,59 @@ import logo from '../lobster-icon.jpg';
 import React from 'react';
 
 export default class Navbar extends React.Component {
+  constructor(props) {
+    super(props);
+    // hard-coded for now, will be fetched from db
+    this.state = {
+      // categoryID of the current category
+      loggedIn: false,
+      authInstance: null,
+    };
+    this.formatUser = this.formatUser.bind(this);
+    this.gapiSetState = this.gapiSetState.bind(this);
+    this.setLoggedIn = this.setLoggedIn.bind(this);
+    this.signOut = this.signOut.bind(this);
+  }
+
+  componentDidMount() {
+    window.gapi.load('auth2', this.gapiSetState);
+  }
+
+  gapiSetState() {
+    window.gapi.auth2.init({
+      client_id: process.env.REACT_APP_CLIENT_ID,
+    });
+    /* Ready. Make a call to gapi.auth2.init or some other API */
+    const authInstance = window.gapi.auth2.getAuthInstance();
+    this.setState({ authInstance });
+    authInstance.isSignedIn.listen(this.setLoggedIn);
+  }
+
+  setLoggedIn(loggedIn) {
+    this.setState({ loggedIn });
+  }
+
+  signOut() {
+    this.state.authInstance.signOut();
```

Comment:
do you need to reset loggedIn to false here?

Reply:
No, as the window.location.reload() reloads the page, and the state defaults to false